### DISCUSSION TITLE
Clarification on refresh_token usage

### DISCUSSION THREAD

**Discussion by reboot** [trakt/trakt-api#316] [Q&A] [answered]
The API documents state that an `access_token` is valid for 3 month "before it needs to be refreshed", which I would interpret as, that you can refresh the token after it has expired. My implementation idea for this was to try to get a new `access_token` when I have a `refresh_token` available and I get a HTTP 401 response from the API, but from by experience so far this does not work.

So I would like to have some clarification on when to use the `refresh_token` to get a new `access_token`. Is it correct that it can only be used as long as the normal `access_token` is valid and if so how long before the `access_token` expires should I try to get a new `access_token`?

**Comment by rectifyer** [comment, marked as answer]
That’s a good question, it should work however the OAuth spec allows but I don’t know what it says specifically. 😄 

@tysonkerridge any insight?

**Reply by tysonkerridge** [reply]
I believe you can use the `refresh_token` at any point to receive a new auth object with a new `access_token`, and perhaps more importantly a new `refresh_token` for the next time you need to refresh. In other words, each time you `POST` to `/oauth/token` with a `refresh_token`, both that `refresh_token` and it's accompanying `access_token` are revoked from my experience.
